src/network/networkd-ndisc.c

```diff
@@ -389,6 +389,35 @@ static int ndisc_router_process_icmp6_ratelimit(Link *link, sd_ndisc_router *rt)
         return 0;
 }
 
+static int ndisc_router_process_retrans_time(Link *link, sd_ndisc_router *rt) {
+        char buf[DECIMAL_STR_MAX(usec_t)];
+        usec_t retrans_time;
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(rt);
+
+        if (!link->network->ipv6_accept_ra_use_retrans_time)
+                return 0;
+
+        r = sd_ndisc_router_get_retrans_time(rt, &retrans_time);
+        if (r < 0) {
+                log_link_debug(link, "Failed to get ICMP6 ratelimit from RA, ignoring: %m");
+                return 0;
+        }
+
+        /* Set the retransmission time for Neigbor Solicitations.
+         * 0 is the unspecified value and must not be set (see RFC4861, 6.3.4) */
+        if (retrans_time != 0) {
```

Comment:
As you can see in the above `ndisc_router_process_icmp6_ratelimit()`, maybe better to use `timestamp_is_set()` and return earlier.

Reply:
FWIW, through testing I found that if I do not use `timestamp_is_set()` then I can still set `infinity` as a valid value.  The value resolves to: `1271310320`.  In any case, `infinity` will be excluded with `timestamp_is_set()` which seems reasonable.

Reply:
After more testing and code reading, I believe we do not need a check for `inifinity` here, since this is the value that comes from the RA and its value is limited to `MAX(uint32_t)`.  So to be pedantic, we should only check for a value of `0` here.  In any case, using `timestamp_is_set()` will cause no harm, so I guess consistency with the `ndisc_router_process_icmp6_ratelimit()` function is king.

_NOTE_: we do need this check in `networkd-sysctl.c:link_set_ipv6_retransmission_time()` which uses the value from the `.network` file, where we can assign `IPv6RetransmissionTimeSec=infinity`.